src/api/common/validate.js

```diff
@@ -24,6 +24,19 @@ function validateAddressAndSecret(obj) {
   }
 }
 
+function validateSecret(secret) {
+  if (!secret) {
+    throw error('Parameter missing: secret');
+  }
+  try {
+    if (!core.Seed.from_json(secret).is_valid()) {
```

Comment:
Yeah, it really doesn't throw anything.

And I've made some changes to Seed.parse_json so it try to parse as different types